packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -262,20 +271,178 @@ gmaps.Size? _gmSizeFromIconConfig(List<Object?> iconConfig, int sizeIndex) {
+      icon.scaledSize = gmaps.Size(targetWidth, targetHeight);
+    }
+  }
+}
+
+/// Fetches the size of the bitmap from a given URL and caches the result.
+///
+/// This method attempts to fetch the image size for a given [url] and caches
+/// the result using [cacheKey]. If the size has already been calculated and
+/// cached for this key, the cached value is returned immediately to improve
+/// performance.
+///
+/// - Parameters:
+/// [url] is the URL of the image for which the size is being fetched.
+/// [cacheKey] is a unique key for each image used for caching the size calculation result.
+Future<Size?> _getBitmapSize(String url, String cacheKey) async {
+  if (_bitmapSizeCompleterCache.containsKey(cacheKey)) {
+    return _bitmapSizeCompleterCache[cacheKey]!.future;
+  }
+  final Completer<Size?> completer = Completer<Size?>();
+  _bitmapSizeCompleterCache[cacheKey] = completer;
+  final HTMLImageElement image = HTMLImageElement()..src = url;
+
+  // Wait for the onLoad or onError event.
+  await Future.any(<Future<Event>>[image.onLoad.first, image.onError.first]);
+
+  if (image.width == 0 || image.height == 0) {
+    // Complete with null for invalid images.
+    completer.complete(null);
+  } else {
+    // Complete with the image size for valid images.
+    completer.complete(Size(image.width.toDouble(), image.height.toDouble()));
+  }
+
+  return completer.future;
+}
+
+/// Cleans up the caches used for bitmap size conversion and URL storage.
+///
+/// This method should be called after marker updates are processed to ensure
+/// that memory usage is optimized by removing completed or outdated cache
+/// entries.
+void _cleanUpBitmapConversionCaches() {
+  _bitmapSizeCompleterCache
+      .removeWhere((_, Completer<Size?> completer) => completer.isCompleted);
+  _bitmapBlobUrlCache.clear();
+}
+
+// Converts a [BitmapDescriptor] into a [gmaps.Icon] that can be used in Markers.
+Future<gmaps.Icon?> _gmIconFromBitmapDescriptor(
+    BitmapDescriptor bitmapDescriptor) async {
   gmaps.Icon? icon;
+  if (bitmapDescriptor is BytesMapBitmap) {
+    final Uint8List bytes = bitmapDescriptor.byteData;
 
+    late String url;
+    if (_bitmapBlobUrlCache.containsKey(bytes.hashCode)) {
+      url = _bitmapBlobUrlCache[bytes.hashCode]!;
+    } else {
+      // TODO(ditman): Improve this conversion
+      // See https://github.com/dart-lang/web/issues/180
+      final Blob blob = Blob(<JSUint8Array>[bytes.toJS].toJS);
+      url = URL.createObjectURL(blob as JSObject);
+      _bitmapBlobUrlCache[bytes.hashCode] = url;
+    }
+
+    icon = gmaps.Icon()..url = url;
+
+    switch (bitmapDescriptor.bitmapScaling) {
+      case MapBitmapScaling.auto:
+        await _setIconSize(
+            width: bitmapDescriptor.width,
+            height: bitmapDescriptor.height,
+            imagePixelRatio: bitmapDescriptor.imagePixelRatio,
+            url: url,
+            cacheKey: bytes.hashCode.toString(),
```

Comment:
The URL of the Blob should be "unique enough" for this cacheKey. I think cacheKey can be removed.